From 6b613b146021268ddac08bd47b310efba8c75ef0 Mon Sep 17 00:00:00 2001
From: Jesse Keating <jkeating@redhat.com>
Date: Fri, 27 May 2011 19:53:34 -0700
Subject: [PATCH 1/2] Handle indented config sections

This brings over some new code from RawConfigParser that handles lines
that start with spaces but are also sections themselves, eg:

[color]
    ui = auto
  [color "branch"]

Previously this would cause a traceback.
(https://bugzilla.redhat.com/show_bug.cgi?id=706218)
---
 lib/git/config.py |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/lib/git/config.py b/lib/git/config.py
index e5fd990..236b4dc 100644
--- a/lib/git/config.py
+++ b/lib/git/config.py
@@ -201,6 +201,11 @@ class GitConfigParser(cp.RawConfigParser, object):
             if line.split(None, 1)[0].lower() == 'rem' and line[0] in "rR":
                 # no leading whitespace
                 continue
+            # continuation line?
+            if line[0].isspace() and cursect is not None and optname:
+                value = line.strip()
+                if value:
+                    cursect[optname] = "%s\n%s" % (cursect[optname], value)
             else:
                 # is it a section header?
                 mo = self.SECTCRE.match(line)
-- 
1.7.4.4

